compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2009,6 +2010,14 @@ public void visit(BLangInvocation invocationExpr) {
         analyzeExprs(invocationExpr.requiredArgs);
         analyzeExprs(invocationExpr.restArgs);
 
+        // Log an error if the function returns never and invoked invalidly.
+        if (invocationExpr.symbol != null &&
+                ((BInvokableSymbol) invocationExpr.symbol).getReturnType().tag == TypeTags.NEVER &&
+                invocationExpr.parent.getKind() != NodeKind.EXPRESSION_STATEMENT) {
+            dlog.error(invocationExpr.pos, DiagnosticCode.INVALID_NEVER_RETURN_TYPED_FUNCTION_INVOCATION,
+                    invocationExpr.name);
+        }
+
         if ((invocationExpr.symbol != null) && invocationExpr.symbol.kind == SymbolKind.FUNCTION) {
```

Comment:
We seem to be repeating the `invocationExpr.symbol != null` check. Can we check if it null and return if so before L2014? Then we can check the rest.

Reply:
There was an error occurred in the integration tests. I changed the behaviour